ir/type.def

```diff
@@ -498,6 +498,17 @@ class Type_Typedef : Type_Declaration, IAnnotated {
 #nodbprint
 }
 
+/// A newtype is like a typedef, but the type does not inherit any
```

Comment:
Additionally, `typedef` being just an alias it could probably be eliminated by the frontend, which is not the case for `type`.